Library/Homebrew/requirements/python_requirement.rb

```diff
@@ -24,7 +24,9 @@ class PythonRequirement < Requirement
       ENV.prepend_path "PATH", Formula["python"].opt_bin
     end
 
-    ENV["PYTHONPATH"] = "#{HOMEBREW_PREFIX}/lib/python#{short_version}/site-packages"
+    if python_binary == "python"
```

Comment:
Maybe it's worth refactoring this into a method and calling it from both pythons `env do` block. I believe that will sort all the problems as well as ensuring it's at the front of the `PATH`.